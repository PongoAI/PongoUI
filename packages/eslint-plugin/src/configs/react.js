/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
