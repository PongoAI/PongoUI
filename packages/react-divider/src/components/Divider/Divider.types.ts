import type { ComponentState, ComponentProps } from '@fluentui/react-utilities';
import type { DividerSlots as FluentDividerSlots } from '@fluentui/react-divider';
type DividerCommons = {
  /**
   * The color of the Divider.
   *
   * @defaultValue inherit
   */
  color?: 'inherit' | 'brand';

  /**
   * Determines the alignment of the content within the divider.
   * @defaultvalue 'center'
   */
  alignContent: 'start' | 'center' | 'end';

  /**
   * A divider can have one of the preset appearances.
   * When not specified, the divider has its default appearance.
   */
  appearance?: 'brand' | 'strong' | 'subtle';

  /**
   * Adds padding to the beginning and end of the divider.
   * @defaultvalue false
   */
  inset: boolean;

  /**
   * A divider can be horizontal (default) or vertical.
   * @defaultvalue false
   */
  vertical: boolean;
};

export type DividerProps = ComponentProps<Partial<FluentDividerSlots>> & Partial<DividerCommons>;

export type DividerState = ComponentState<FluentDividerSlots> & DividerCommons;
