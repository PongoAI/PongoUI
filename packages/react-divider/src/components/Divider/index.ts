export * from './Divider';
export * from './Divider.types';
export * from './useDividerStyles';
