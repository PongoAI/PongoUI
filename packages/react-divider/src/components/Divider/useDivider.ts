import * as React from 'react';
import type { DividerProps, DividerState } from './Divider.types';
import { useDivider_unstable } from '@fluentui/react-divider';

/**
 * Returns the props and state required to render the component
 * @param props - User-provided props to the Divider component.
 * @param ref - User-provided ref to be passed to the Divider component.
 */
export const useDivider = (props: DividerProps, ref: React.Ref<HTMLElement>): DividerState => {
  const { color = 'inherit' } = props;

  const state = useDivider_unstable(props, ref);

  return {
    color,
    ...state,
  };
};
