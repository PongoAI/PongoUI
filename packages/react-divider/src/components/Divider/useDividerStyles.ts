import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { DividerState } from './Divider.types';
import { tokens } from '@pongo-ui/react-theme';

const contentSpacing = '12px';
const insetSpacing = '12px';
const startEndMaxHeight = '8px';

const useRootStyles = makeStyles({
  root: {
    position: 'relative',
    alignItems: 'center',
    boxSizing: 'border-box',
    display: 'flex',
    flexDirection: 'row',
    flexGrow: 1,
    fontFamily: tokens.baseFont,
    fontSize: tokens.fontSize200,
    fontWeight: tokens.regular,
    lineHeight: tokens.lineHeight200,

    ':before': {
      boxSizing: 'border-box',
      display: 'flex',
      flexGrow: 1,
    },

    ':after': {
      boxSizing: 'border-box',
      display: 'flex',
      flexGrow: 1,
    },
  },

  childless: {
    ':before': {
      marginBottom: 0,
      marginRight: 0,
    },

    ':after': {
      marginLeft: 0,
      marginTop: 0,
    },
  },

  inset: {
    marginTop: insetSpacing,
    marginBottom: insetSpacing,
  },

  // Positioning
  start: {
    ':after': {
      content: '""',
    },
  },
  center: {
    ':before': {
      content: '""',
    },
    ':after': {
      content: '""',
    },
  },
  end: {
    ':before': {
      content: '""',
    },
  },

  // Color
  inherit: {
    color: tokens.inherit,

    ':before': {
      ...shorthands.borderColor(tokens.inheritBackground),
    },
    ':after': {
      ...shorthands.borderColor(tokens.inheritBackground),
    },
  },

  brand: {
    color: tokens.brand,

    ':before': {
      ...shorthands.borderColor(tokens.brandBackground),
    },
    ':after': {
      ...shorthands.borderColor(tokens.brandBackground),
    },
  },
});

const useHorizontalStyles = makeStyles({
  base: {
    width: '100%',

    ':before': {
      borderTopStyle: 'solid',
      borderTopWidth: '1px',
    },

    ':after': {
      borderTopStyle: 'solid',
      borderTopWidth: '1px',
    },
  },

  inset: {
    paddingLeft: insetSpacing,
    paddingRight: insetSpacing,
  },

  start: {
    ':before': {
      content: '""',
      marginRight: contentSpacing,
      maxWidth: startEndMaxHeight,
    },

    ':after': {
      marginLeft: contentSpacing,
    },
  },
  center: {
    ':before': {
      marginRight: contentSpacing,
    },
    ':after': {
      marginLeft: contentSpacing,
    },
  },
  end: {
    ':before': {
      marginRight: contentSpacing,
    },
    ':after': {
      content: '""',
      marginLeft: contentSpacing,
      maxWidth: startEndMaxHeight,
    },
  },
});

const useVerticalStyles = makeStyles({
  base: {
    flexDirection: 'column',
    minHeight: '20px',

    ':before': {
      borderRightStyle: 'solid',
      borderRightWidth: '1px',
    },

    ':after': {
      borderRightStyle: 'solid',
      borderRightWidth: '1px',
    },
  },

  inset: {
    marginTop: insetSpacing,
    marginBottom: insetSpacing,
  },

  withChildren: {
    minHeight: '84px',
  },

  start: {
    ':before': {
      content: '""',
      marginBottom: contentSpacing,
      maxHeight: startEndMaxHeight,
    },

    ':after': {
      marginTop: contentSpacing,
    },
  },
  center: {
    ':before': {
      marginBottom: contentSpacing,
    },
    ':after': {
      marginTop: contentSpacing,
    },
  },
  end: {
    ':before': {
      marginBottom: contentSpacing,
    },
    ':after': {
      content: '""',
      marginTop: contentSpacing,
      maxHeight: startEndMaxHeight,
    },
  },
});

export const useDividerStyles = (state: DividerState) => {
  const { color, inset, vertical, alignContent } = state;
  const rootStyles = useRootStyles();
  const horizontalStyles = useHorizontalStyles();
  const verticalStyles = useVerticalStyles();

  state.root.className = mergeClasses(
    rootStyles.root,
    rootStyles[alignContent!],

    // Horizontal styles
    !vertical && horizontalStyles.base,
    !vertical && inset && horizontalStyles.inset,
    !vertical && horizontalStyles[alignContent!],

    // Vertical styles
    vertical && verticalStyles.base,
    vertical && inset && verticalStyles.inset,
    vertical && verticalStyles[alignContent!],
    vertical && state.root.children !== undefined && verticalStyles.withChildren,

    // Children
    state.root.children === undefined && rootStyles.childless,

    // Color
    rootStyles[color!],

    state.root.className,
  );

  return state;
};
