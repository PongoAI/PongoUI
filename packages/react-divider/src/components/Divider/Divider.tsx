import * as React from 'react';
import { renderDivider_unstable } from '@fluentui/react-divider';
import { useDivider } from './useDivider';
import { useDividerStyles } from './useDividerStyles';
import type { DividerProps } from './Divider.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Divider helps separate content in a site.
 */
export const Divider: ForwardRefComponent<DividerProps> = React.forwardRef((props, ref) => {
  const state = useDivider(props, ref);

  useDividerStyles(state);

  return renderDivider_unstable(state);
});
Divider.displayName = 'Divider';
