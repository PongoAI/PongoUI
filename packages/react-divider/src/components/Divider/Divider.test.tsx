import React from 'react';
import { Divider } from './Divider';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';
import { resetIdsForTests } from '@fluentui/react-utilities';

describe('Divider', () => {
  isConformant({
    Component: Divider,
    displayName: 'Divider',
  });

  beforeEach(() => {
    resetIdsForTests();
  });

  it('renders a divider', () => {
    const result = render(<Divider />);

    expect(result.baseElement).toMatchSnapshot();
  });

  it('renders a divider with content', () => {
    const result = render(<Divider>This is a test</Divider>);

    expect(result.baseElement).toMatchSnapshot();
  });
});
