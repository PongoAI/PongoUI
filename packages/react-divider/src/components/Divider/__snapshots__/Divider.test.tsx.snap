// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders a divider 1`] = `
<body>
  <div>
    <div
      aria-orientation="horizontal"
      class="___zwr68b0 f10pi13n f122n59 f1ewtqcl f22iagw f1063pyq fqerorx f8dqhol f15hzwe6 f1rd5vql f8s6rei f170vdtw fkmkm9e fbxxcv3 ffdc0f3 fsnc50s fjrry2l fuzzvh5 fqc6z8f fly5x3f f1bwkb58 f199l62q f1iq3fds f1l5cenb fhlqc5k f1957mjw fc3uzr1 f1hmx8el f1ysjbbf f6dxpfp fghvtis ftbb52h f1ohrftm f1jjt28w f1tz0uvw f11imphh f14js4i3"
      role="separator"
    />
  </div>
</body>
`;

exports[`Divider renders a divider with content 1`] = `
<body>
  <div>
    <div
      aria-labelledby="divider-1"
      aria-orientation="horizontal"
      class="___nz8i480 f10pi13n f122n59 f1ewtqcl f22iagw f1063pyq fqerorx f8dqhol f15hzwe6 f1rd5vql f8s6rei f170vdtw fkmkm9e fbxxcv3 ffdc0f3 fsnc50s fjrry2l fuzzvh5 fqc6z8f fly5x3f f1bwkb58 f199l62q f1iq3fds f1l5cenb fncq426 f1io6r2y f1ysjbbf f6dxpfp fghvtis ftbb52h f1ohrftm f1jjt28w f1tz0uvw f11imphh f14js4i3"
      role="separator"
    >
      <div
        id="divider-1"
      >
        This is a test
      </div>
    </div>
  </div>
</body>
`;
